drivers/reset/reset_mchp_mss.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2025 embedded brains GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_mpfs_reset
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/reset.h>
+
+#define SUBBLK_CLOCK_CR_OFFSET 0x84U
+
+#define SOFT_RESET_CR_OFFSET 0x88U
+
+#define RESET_MSS_REG_BIT(id) ((id) & 0x1fU)
+
+struct reset_mss_config {
+	uintptr_t base;
+};
+
+static int reset_mss_status(const struct device *dev, uint32_t id, uint8_t *status)
+{
+	const struct reset_mss_config *config = dev->config;
+
+	/* Device is in reset if the clock is turned off or held in soft reset */
+	*status = sys_test_bit(config->base + SUBBLK_CLOCK_CR_OFFSET, RESET_MSS_REG_BIT(id)) == 0 ||
+		  sys_test_bit(config->base + SOFT_RESET_CR_OFFSET, RESET_MSS_REG_BIT(id) != 0);
+
+	return 0;
+}
+
+static int reset_mss_line_assert(const struct device *dev, uint32_t id)
+{
+	const struct reset_mss_config *config = dev->config;
+
+	/* Turn off clock */
+	sys_clear_bit(config->base + SUBBLK_CLOCK_CR_OFFSET, RESET_MSS_REG_BIT(id));
```

Comment:
bit 17 (RESERVED) of the `SUBBLK_CLOCK_CR` register is RO from the PolarFire SoC regmap, writing to it can result in indeterminate events.

Reply:
I replaced the sys_test_bit() with IS_BIT_SET().